### ISSUE TITLE
[Infrastructure/Testing] Stage a deploy on PRs

### ISSUE THREAD

**Issue by parlough** [dart-lang/dartdoc#2513]
I'm not really sure if there's a nice, easy way to do this, but it would be really cool if we could somehow stage the generated docs for the Dart SDK and/or Flutter in a PR. Perhaps only for collaborators or those who have committed before.

**Comment by jcollins-g** [comment]
@parlough Ah, basically, something like grind serve-test-package-docs and friends run automatically so we could browse or refer to output changes in PRs?  That would be nice.

**Comment by parlough** [comment]
Yeah that's what I'm thinking, I think Flutter would likely be the most useful, but really any larger library would be great. Github actions can build the docs, but we'd need to figure out some way to host them. Options could include Firebase hosting, Cloud storage?, or maybe Github pages. I'm not really sure though and can investigate more in the future.

**Comment by parlough** [comment]
@athomas I wonder if you would have any ideas around this idea after converting `site-www`'s deploy to Github actions.

**Comment by athomas** [comment]
I looked a bit at the firebase github integration when migrating dart.dev, but it requires an exported service account key which would need some extra approvals internally:
https://firebase.google.com/docs/hosting/github-integration
